vortex-io/src/read.rs

```diff
@@ -73,7 +75,7 @@ impl VortexReadAt for Buffer {
                 vortex_err!("unexpected eof"),
             )))
         } else {
-            let mut buffer = BytesMut::with_capacity(len.try_into().vortex_unwrap());
+            let mut buffer = VortexBytesMut::with_capacity(len.try_into().vortex_unwrap());
```

Comment:
I'm hoping this fixes it!

Reply:
Hmm, I forget why this is this even copying instead of just slicing?

Reply:
Oh bc buffer and bytes. Can we just use the new fron_owner and make a Bytes from Buffer directly